src/utils/helpers/structure/addApplicationResponses.ts

```diff
@@ -0,0 +1,30 @@
+import { ApplicationResponse } from '../../generated/graphql'
+import { FullStructure } from '../../types'
+
+const addApplicationResponses = (
+  structure: FullStructure,
+  applicationResponses: ApplicationResponse[]
+) => {
+  const templateElementIsFirst: { [key: number]: boolean } = {}
+  // Application response are ordered by timestamps, so we can be sure that second response with same templateElementId is previousApplicationResponse
+  applicationResponses.forEach((applicationResponse) => {
+    const element = structure?.elementsById?.[applicationResponse?.templateElementId || '']
+    const templateElementId = applicationResponse?.templateElementId || 0
+    // if templateElement id is not found in templateElementIsFirst object, it's the first one for templateElementId
+    if (!(templateElementId in templateElementIsFirst)) {
+      templateElementIsFirst[templateElementId] = true
+      // "value" is not available in element.response (at least not in the FullResponse type itself)
+      // thus addition of latestApplicationResponse to be used in addChangeRequesStatusToElement
```

Comment:
ok, it will be done via this discussion https://github.com/openmsupply/application-manager-web-app/issues/456

and removed comment: 30c825d3a8544b8575b04c03f23547fb8ef52992